core/src/main/java/bio/terra/pearl/core/service/workflow/RegistrationService.java

```diff
@@ -125,6 +137,33 @@ private RegistrationResult register(String portalShortcode, EnvironmentName envi
         return new RegistrationResult(user, ppUser);
     }
 
+    public RegistrationResult registerGovernedUser(String portalShortcode, ParticipantUser proxy) {
+        PortalEnvironment portalEnv = portalEnvService.findOne(portalShortcode, proxy.getEnvironmentName()).get();
+        ParticipantUser governedUser = new ParticipantUser();
+        governedUser.setEnvironmentName(proxy.getEnvironmentName());
+        String governedUsernameSuffix = generateGovernedUsernameSuffix(proxy.getUsername(), proxy.getEnvironmentName());
+        String governedUserName = GOVERNED_USERNAME_INDICATOR.formatted(proxy.getUsername(), governedUsernameSuffix);//a@b.com-prox-guid
+        governedUser.setUsername(governedUserName);
+        governedUser = participantUserService.create(governedUser);
+
+        PortalParticipantUser governedPpUser = new PortalParticipantUser();
+        governedPpUser.setPortalEnvironmentId(portalEnv.getId());
+        governedPpUser.setParticipantUserId(governedUser.getId());
+
+        Profile governedProfile =  Profile.builder()
+                .contactEmail(proxy.getUsername())
+                .givenName(null)
+                .familyName(null)
+           .build();
```

Comment:
We want to use the contact email, not the username.  this will cover the case where the proxy has changed their contact email, and then registers a governed user. (e.g. a parent registers a kid, changes address, then registers another kid)
```suggestion
        PortalParticipantUser proxyPpUser = portalParticipantUserService.findOne(proxy.getId(), portalShortcode).orElseThrow(
                () -> new NotFoundException("Portal user not found for proxy in portal %s".formatted(portalShortcode))
        );
        Profile proxyProfile = profileService.find(proxyPpUser.getProfileId()).orElseThrow();
        Profile governedProfile =  Profile.builder()
                .contactEmail(proxyProfile.getContactEmail())
                .givenName(null)
                .familyName(null)
           .build();
```

Reply:
Ah! great catch!